buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy

```diff
@@ -233,6 +233,8 @@ class BeamModulePlugin implements Plugin<Project> {
 
     // Required. Pipeline options to be used by the tested pipeline.
     String integrationTestPipelineOptions = System.getProperty('integrationTestPipelineOptions')
+
+    String experiment = System.getProperty('experiment', '')
```

Comment:
Adding an 'experiment' flag here could be confused with the pipelineOptions experiment flag. Maybe instead of this, it would be better to inspect the pipelineOptions instead?